secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift

```diff
@@ -1,11 +1,146 @@
+            }
             .padding()
-            .navigationTitle("Transaction: \(transaction.id)")
+
+            Text("\(amountPrefixText) \(transaction.zecAmount.decimalString()) ZEC")
+                .transactionDetailRow()
+            
+            Text("fee \(transaction.fee.decimalString()) ZEC")
+                .transactionDetailRow()
+
+            Text("total amount \(transaction.totalAmount.decimalString()) ZEC")
+                .transactionDetailRow()
+
+            Button {
+                viewStore.send(.copyToPastboard(transaction.address))
+            } label: {
+                Text("\(addressPrefixText) \(transaction.address)")
+                    .lineLimit(1)
+                    .truncationMode(.middle)
+                    .transactionDetailRow()
+            }
+
+            if let memo = transaction.memo {
+                Button {
+                    viewStore.send(.copyToPastboard(memo))
+                } label: {
+                    VStack {
+                        Text("\(memo)")
+                            .multilineTextAlignment(.leading)
+                        
+                        HStack {
+                            Text("reply-to address included")
+                            Spacer()
+                            Button {
+                                viewStore.send(.replyTo(transaction.address))
+                            } label: {
+                                Text("reply now")
+                                    .padding(5)
+                                    .overlay(
+                                        RoundedRectangle(cornerRadius: 6)
+                                            .stroke(Asset.Colors.Text.transactionDetailText.color, lineWidth: 1)
+                                    )
+                            }
+                        }
+                    }
+                    .transactionDetailRow()
+                }
+            }
+
+            HStack {
+                Text("Confirmed")
+                Spacer()
+                Text("\(transaction.confirmations) times")
+            }
+            .transactionDetailRow()
+
+            Spacer()
+            
+            Button {
+                viewStore.send(.copyToPastboard(transaction.id))
+            } label: {
+                Text("txn:\(transaction.id)")
+                    .foregroundColor(Asset.Colors.Text.transactionDetailText.color)
+                    .font(.system(size: 14))
+                    .fontWeight(.thin)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(.horizontal, 60)
+                    .padding(.vertical, 10)
+                    .background(Asset.Colors.BackgroundColors.numberedChip.color)
+                    .padding(.vertical, 30)
+            }
+
+            Button {
+                viewStore.send(.viewTxidOnline)
+            } label: {
+                if let viewOnlineURL = transaction.viewOnlineURL {
```

Comment:
```suggestion
                // TODO: Warn users that they will leave the App when they follow a Block explorer
                // https://github.com/zcash/secant-ios-wallet/issues/379
                if let viewOnlineURL = transaction.viewOnlineURL {
```

Reply:
Good point, added the comment. 

Small comment: SwiftUI's `Link` is used to go outside the app. It's not in our control, it's not done in the TCA way. I'll take some time to figure out how to do this better way so we gain some control.